src/Psl/Str/Grapheme/length.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Str\Grapheme;
+
+use function grapheme_strlen;
+
+/**
+ * Returns the length of the given string in grapheme units
+ *
+ * @psalm-pure
+ */
+function length(string $string): int
+{
+    return (int) grapheme_strlen($string);
```

Comment:
I think there's a case where grapheme_strlen returns `null`, not sure how to handle that.

Reply:
The docs state it always returns an int though:
https://www.php.net/manual/en/function.grapheme-strlen.php

(We could keep it - just to be sure :) )

Reply:
I learned to never belive the documentation 😆  

see:

- https://github.com/php/php-src/blob/e73bf01f9d0c23dd5baed4445b37c93ec4e0ab29/ext/intl/php_intl.stub.php#L246
- https://github.com/php/php-src/blob/7af24eaedd310911940643d7a5f412393b973389/ext/intl/grapheme/grapheme_string.c#L84
- https://github.com/php/php-src/blob/7af24eaedd310911940643d7a5f412393b973389/ext/intl/grapheme/grapheme_string.c#L96

not sure how to handle these cases ...

Reply:
I added invariant violation checks.